### DISCUSSION TITLE
Unable to commit & sync from Codespaces due to "divergent branches"

### DISCUSSION THREAD

**Discussion by cmq879** [earthlab-education/Earth-Analytics-2023-01-Intro#383] [! GitHub help] [answered]
Been attempting to commit an updated notebook from Codespaces but continually getting the following error. My repo has only a main branch; I don't know what any of this means. 

> git pull --tags origin main
From https://github.com/earthlab-education/reproducible-science-cmq879
 * branch            main       -> FETCH_HEAD
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge (the default strategy)
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.

**Comment by ellenalamont17** [comment, marked as answer]
There is a small clip near the end of the [Uploading an analysis to your portfolio demo video](https://cuboulder.zoom.us/rec/share/SG7RTxqDJuNfA5NzkvbmqmSXXE1AezrnS6R3KfW7M3atoXBdJy9cw4bsa-jj6fqQ.QNYFMmssw-4CCApC) that covers how to deal with this.

Basically, you need to do the following in the terminal:
1. Navigate to your site directory. For example, I would type: cd ellenalamont17.github.io
2. Then type: git config --global pull.rebase

These two things should allow you to commit the changes. In your codespace, you can find the terminal here:
![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143370562/78e97c08-36a8-41b5-b1be-e853f0168636)

**Reply by eculler** [reply]
Just a small tweak! The command is `git config --global pull.rebase false`

**Comment by earthlab-test-student** [comment]
Hey folks - an update on this! You can actually get Sync to work from the menu, if you do the following:

1. Go to the Source Control tab
2. Click the ... in the upper right of that column
3. Select Pull > Pull (rebase)
4. Now you will be able to Sync
